### ISSUE TITLE
Compiling druntime/Phobos with LTO fails for Windows

### ISSUE THREAD

**Issue by kinke** [ldc-developers/ldc#2755]
The following modules fail to compile with `-flto=thin` (full too IIRC) on Win64:

* std.internal.digest.sha_SSSE3
* core.bitop
* ldc.eh_msvc
* core.sys.windows.threadaux
* core.thread

due to an LLVM unreachable with LLVM 6.0.0:

```
this directive only supported on COFF targets
UNREACHABLE executed at C:\LDC\llvm\lib\MC\MCStreamer.cpp:915!
```

AFAICT, those are the modules with `naked` inline asm functions.

**Comment by JohanEngelen** [comment]
Simple naked asm works (https://github.com/ldc-developers/ldc/blob/master/tests/linking/thinlto_asm_x86.d is run on Appveyor, right?), but perhaps more complex cases don't.

**Comment by kinke** [comment]
Yep, LLVM unreachable hit when compiling `tests/linking/inputs/asm_x86.d` with thin/full LTO, for both Win32 and Win64.
